hs_swat_modelinstance/page_processors.py

```diff
@@ -0,0 +1,125 @@
+__author__ = 'Mohamed Morsy'
+from mezzanine.pages.page_processors import processor_for
+from crispy_forms.layout import Layout, Fieldset, ButtonHolder, Submit, HTML
+from hs_core import page_processors
+from hs_core.views import *
+from forms import *
+from hs_swat_modelinstance.models import SWATModelInstanceResource
+
+@processor_for(SWATModelInstanceResource)
+def landing_page(request, page):
+    content_model = page.get_content_model()
+    edit_resource = page_processors.check_resource_mode(request)
+
+    if not edit_resource:
+        # get the context from hs_core
+        context = page_processors.get_page_context(page, request.user, resource_edit=edit_resource, extended_metadata_layout=None)
+        extended_metadata_exists = False
+        if content_model.metadata.model_output or \
+                content_model.metadata.executed_by or \
+                content_model.metadata.model_objective or \
+                content_model.metadata.simulation_type or \
+                content_model.metadata.model_methods or \
+                content_model.metadata.swat_model_parameters or \
+                content_model.metadata.model_input:
+            extended_metadata_exists = True
+
+        context['extended_metadata_exists'] = extended_metadata_exists
+        context['model_output'] = content_model.metadata.model_output
+        context['executed_by'] = content_model.metadata.executed_by
+        context['model_objective'] = content_model.metadata.model_objective
+        context['simulation_type'] = content_model.metadata.simulation_type
+        context['model_methods'] = content_model.metadata.model_methods
+        context['swat_model_parameters'] = content_model.metadata.swat_model_parameters
+        context['model_input'] = content_model.metadata.model_input
```

Comment:
I'd recommend making this all simplier by looping over the names and and for each setting `extended_metadata_exists` if needed and injecting into `context`